src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs

```diff
@@ -56,7 +56,7 @@ public sealed override void Initialize(AnalysisContext context)
                     return;
                 }
 
-                var membersByValue = new ConcurrentDictionary<object, ConcurrentBag<(SyntaxNode fieldSyntax, string fieldName)>>();
+                var membersByValue = new ConcurrentDictionary<object, ConcurrentDictionary<(SyntaxNode fieldSyntax, string fieldName), Unit>>();
```

Comment:
I don't recall if `GetLocation` is an expensive call, if not maybe we could directly store the location span rather than the `SyntaxNode`.